### ISSUE TITLE
Exception thrown while parsing command line arguments with dotnet new

### ISSUE THREAD

**Issue by MichaelSimons** [dotnet/cli#5470]
## Steps to reproduce
`dotnet new Web1.1`
e.g. forgot the `-t`

## Expected  behavior
I expect some output that would indicate the specified arguments were invalid possibly even listing what part was invalid.  Possibly even display the --help content.  I do not expect to see an unhandled exception

## Actual behavior

```
Specify --help for a list of available options and commands.

Unhandled Exception: Microsoft.DotNet.Cli.CommandLine.CommandParsingException: Unrecognized command or argument 'Web1.1'
   at Microsoft.DotNet.Cli.CommandLine.CommandLineApplication.HandleUnexpectedArg(CommandLineApplication command, String[] args, Int32 index, String argTypeName)
   at Microsoft.DotNet.Cli.CommandLine.CommandLineApplication.Execute(String[] args)
   at Microsoft.DotNet.Tools.New.NewCommand.Run(String[] args)
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, ITelemetry telemetryClient)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)
```

## Environment data
`dotnet --info` output:

```
.NET Command Line Tools (1.0.0-rc4-004616)

Product Information:
 Version:            1.0.0-rc4-004616
 Commit SHA-1 hash:  56d8071361

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\1.0.0-rc4-004616
```

**Comment by dasMulli** [comment]
Looks like `new3` handles it well and new will be replaced by it eventually. But unhandled exceptions are very bad on macOS!:
<img width="830" alt="screen shot 2017-01-26 at 20 38 17" src="https://cloud.githubusercontent.com/assets/6129250/22347236/86b365f0-e407-11e6-8f17-1028355bc5d2.png">

**Comment by piotrpMSFT** [comment]
Fixed by #5493
